File: javascript/grade-school/grade-school.js

export class GradeSchool {
  constructor(){
    this.roster_ ={};
  }
  roster() {
    return Object.keys(this.roster_).reduce((res, g) => {
      res[g] = [...this.roster_[g]];
      return res;
    },{});
  }

  add(student, grade) {
    if(!this.roster_[grade]){
      this.roster_[grade] =[]
    }
    this.roster_[grade].push(student);
    this.roster_[grade].sort();
  }

  grade(g) {
    return g in this.roster_ ? [...this.roster_[g]] : [];
  }
}

File: lua/grade-school/grade-school.lua
local school ={}
school.__index =school


function school:new()
	local self = setmetatable({rost ={},}, school)
	return self
end

function school:add( name, grade)
	self.rost[grade] = self.rost[grade] or {}
	table.insert(self.rost[grade], name)
	table.sort(self.rost[grade])
	return self
end

function school:grade(g)
	return self.rost[g] or {}
end

function school:roster()
	return self.rost
end

return school
